config.go

```diff
@@ -161,51 +193,75 @@ func convertAndSetSlice(slicePtr interface{}, values []string) {
 // Slice and struct are handled elsewhere.
 // Unhandled kinds panic.
 // Errors in string conversion are ignored, and the settable remains a zero value.
-func convertAndSetValue(settable interface{}, s string) {
+func convertAndSetValue(settable interface{}, s string) (err error) {
 	settableValue := reflect.ValueOf(settable).Elem()
 	switch settableValue.Kind() {
 	case reflect.String:
 		settableValue.SetString(s)
 	case reflect.Int:
-		val, _ := strconv.ParseInt(s, 10, 0)
-		settableValue.SetInt(val)
+		return convertAndSetInt(settableValue, s, 0)
 	case reflect.Int8:
-		val, _ := strconv.ParseInt(s, 10, 8)
-		settableValue.SetInt(val)
+		return convertAndSetInt(settableValue, s, 8)
 	case reflect.Int16:
-		val, _ := strconv.ParseInt(s, 10, 26)
-		settableValue.SetInt(val)
+		return convertAndSetInt(settableValue, s, 26)
 	case reflect.Int32:
-		val, _ := strconv.ParseInt(s, 10, 32)
-		settableValue.SetInt(val)
+		return convertAndSetInt(settableValue, s, 32)
 	case reflect.Int64:
-		val, _ := strconv.ParseInt(s, 10, 64)
-		settableValue.SetInt(val)
+		return convertAndSetInt(settableValue, s, 64)
 	case reflect.Uint:
-		val, _ := strconv.ParseUint(s, 10, 0)
-		settableValue.SetUint(val)
+		return convertAndSetUint(settableValue, s, 0)
 	case reflect.Uint8:
-		val, _ := strconv.ParseUint(s, 10, 8)
-		settableValue.SetUint(val)
+		return convertAndSetUint(settableValue, s, 8)
 	case reflect.Uint16:
-		val, _ := strconv.ParseUint(s, 10, 16)
-		settableValue.SetUint(val)
+		return convertAndSetUint(settableValue, s, 16)
 	case reflect.Uint32:
-		val, _ := strconv.ParseUint(s, 10, 32)
-		settableValue.SetUint(val)
+		return convertAndSetUint(settableValue, s, 32)
 	case reflect.Uint64:
-		val, _ := strconv.ParseUint(s, 10, 64)
-		settableValue.SetUint(val)
+		return convertAndSetUint(settableValue, s, 64)
 	case reflect.Bool:
-		val, _ := strconv.ParseBool(s)
-		settableValue.SetBool(val)
+		return convertAndSetBool(settableValue, s)
 	case reflect.Float32:
-		val, _ := strconv.ParseFloat(s, 32)
-		settableValue.SetFloat(val)
+		return convertAndSetFloat(settableValue, s, 32)
 	case reflect.Float64:
-		val, _ := strconv.ParseFloat(s, 64)
-		settableValue.SetFloat(val)
+		return convertAndSetFloat(settableValue, s, 64)
 	default:
-		panic(fmt.Sprintf("cannot handle kind %v\n", settableValue.Type().Kind()))
```

Comment:
this panic should stay. It will only ever happen with programmer error.

Reply:
that's fair, plenty of examples of this in go :D